docs/swagger-ui/src/main/scala/sttp/tapir/swagger/SwaggerUI.scala

```diff
@@ -20,18 +20,23 @@ object SwaggerUI {
     * @param yaml
     *   The yaml with the OpenAPI documentation.
     * @param prefix
-    *   The path prefix from which the documentation will be served, as a list of path segments. Defaults to `List(docs)`, so the address of
-    *   the docs will be `/docs`.
+    *   The path prefix for which the documentation endpoint will be created, as a list of path segments. Defaults to `List(docs)`, so the
+    *   address of the docs will be `/docs` (unless <code>basePrefix</code> is specified)
     * @param yamlName
     *   The name of the file, through which the yaml documentation will be served. Defaults to `docs.yaml`.
+    * @param basePrefix
+    *   The base path prefix where the documentation routes are going to be attached. Unless the endpoint will served from `/`, the base
+    *   path prefix must be specified for redirect to work correctly. Defaults to `Nil`, so it is assumed that the endpoint base path is
+    *   `/`.
     */
   def apply[F[_]](
       yaml: String,
       prefix: List[String] = List("docs"),
-      yamlName: String = "docs.yaml"
+      yamlName: String = "docs.yaml",
+      basePrefix: List[String] = Nil
   ): List[ServerEndpoint[Any, F]] = {
     val prefixInput: EndpointInput[Unit] = prefix.map(stringToPath).reduce[EndpointInput[Unit]](_.and(_))
-    val prefixAsPath = prefix.mkString("/")
+    val prefixAsPath = (basePrefix ++ prefix).mkString("/")
```

Comment:
the name is misleading - esp when contrasted with `prefixInput`, which uses only `prefix`. So maybe sth like `wholePrefix`?

Reply:
Or `prefixForRedirectAsPath` - as this is only used in the context of redirects if I'm right